examples/acoustic/wavesolver.py

```diff
@@ -108,9 +108,15 @@ def forward(self, src=None, rec=None, u=None, m=None, save=False, **kwargs):
 
         # Execute operator and return wavefield and receiver data
         if save:
-            summary = self.op_fwd_save.apply(src=src, rec=rec, u=u, m=m, **kwargs)
+            summary = self.op_fwd_save.apply(src=src, rec=rec, u=u, m=m,
+                                             src_coords=src.coordinates,
+                                             rec_coords=rec.coordinates,
+                                             **kwargs)
         else:
-            summary = self.op_fwd.apply(src=src, rec=rec, u=u, m=m, **kwargs)
+            summary = self.op_fwd.apply(src=src, rec=rec, u=u, m=m,
+                                        src_coords=src.coordinates,
+                                        rec_coords=rec.coordinates,
```

Comment:
Why do you have to pass the coordinates explicitly when you could derive the directly in the callee from the actual parameters, ie `src` and `rec` ?